### ISSUE TITLE
CAS request for CPU: long running increase

### ISSUE THREAD

**Issue by Maralsotoudehnia** [BCDevOps/devops-requests#523]
## Step 0
**Are you the product owner or technical lead?** Yes, I'm the product owner and @matthieu-foucault is the technical lead.
If not, you will need to ask one of those two people to make this request.

## Step 1
Make sure no duplicated request exists, search here:
https://github.com/BCDevOps/devops-requests/issues

## Step 2
What quotas need to be changed? (Select all that apply)
- [ X] CPU: Long Running (compute-long-running-quota)
   - [ ] CPU Limit
   - [ ] Memory Limit
   - [ ] CPU Request
   - [ ] Memory Request
- [ ] CPU: Time Bound (compute-time-bound-quota)
   - [ ] CPU Limit
   - [ ] Memory Limit
   - [ ] CPU Request
   - [ ] Memory Request
- [ ] Storage: Netapp Block (storage-netapp-block-standard-quota)
   - [ ] Number of PVCs
   - [ ] Total Storage
- [ ] Storage: Netapp File (storage-netapp-file-standard-quota)
   - [ ] Number of PVCs
   - [ ] Total Storage
- [ ] Storage: NFS Backup (storage-nfs-backup-quota)
   - [ ] Number of PVCs
   - [ ] Total Storage
- [ ] Storage: Gluster Block (storage-gluster-block-quota)
   - [ ] Number of PVCs
   - [ ] Total Storage
- [ ] Storage: Gluster File (storage-gluster-file-quota)
   - [ ] Number of PVCs
   - [ ] Total Storage
- [ ] General Storage Quota (storage-quota)
   - [ ] Number of PVCs
   - [ ] Total Storage

## Step 3
Please provide the namespace(s) to which these quota changes need to be applied:
wksv3k-prod

## Step 4
Add details about your quota request here.

Please increase our quota to 32 cores.

## Step 5
Please explain here the reasons for requesting your quota change.

 Our patroni cluster needs more resources to handle the requests. Thank you!

Remember that OCP is a shared resource. Please be community-minded in making these requests and include thorough details about why your project's current resource usage cannot be reduced to adhere to the existing quota requirements.

**Comment by sbarre-esit** [comment]
@jefkel would appreciate your input here. The patroni master takes all the traffic, so all pods in the StatefulSet need a high limit which then eats a lot of quota. I'm not sure if there is a better solution.

[RocketChat discussion](https://chat.pathfinder.gov.bc.ca/channel/devops-sos?msg=B8vcEHir4fRCsn38c)

**Comment by jefkel** [comment]
From the rc conversation, it looks like this is may not be a long term (permanent) increase?  If this is only required for a temporary coverage, what is the target date to resolve the capacity issue?

If this is to be a permanent request, then there's not enough information to be able to appropriately evaluate the request (from a technical perspective).
- have capacity testing/projections been done?  (is there additional growth projected, and what's the expected growth timeline?)
- the quota request is for a namespace, not for a specific deployment, and I'd love to see the request reflect the overall project quota requirements for the full context of the request.
- are links available to documentation (diagrams incl?) of the micro-services and their planned resource usage?

There are some comments about number of stateful set pods that I'd love to see the architecture plans for before any comments/questions about resource utilization are made.

**Comment by Maralsotoudehnia** [comment]
Hi @jefkel: 

This is indeed temporary until end of week. We've had a significant increase in user traffic concentrated today and hit limits. 

Notably, we didn't load test our application because DevOps (at the time) advised me against doing so when I raised the question in 2019.  So, I'd like to clarify that we proceeded in accordance to the guidance given and are not going rogue, here.  I'd appreciate the support I'm requesting to be unblocked immediately as our users are currently unable to complete their applications. 

I can't respond to growth projection as this is 1) a new application, and 2) in support of a new CleanBC program. What we do know is that, barring any changes to the program area, we shouldn't see this sort of user traffic again until late spring of 2021. That said, given the infancy of this program, any assessment is prefatory at best.

I can get update this issue with the additional information you're requesting for quota information, but seeing as it's end of day, I won't be able to get that until tomorrow. 

Again, I'll reiterate that I'd appreciate immediate support relating to this request. I'm happy to debrief after the fact, but this request has some urgency to it. Feel free to contact me directly with any specific questions.

**Comment by jefkel** [comment]
Thanks for the quick response Maral

I believe @sbarre-esit has already implemented an immediate fix, and this was the followup/debrief documentation of that request?  If there is additional emergency response required, please continue that discussion in the RC channel.

The additional information requested was if this was to be a permanent increase in resource quota.  For the temporary increase, I was mainly looking for a projected date to roll-back the quota change (to be confirmed before any rollback of course).

I'd love to have a chat (outside of this ticket) about performance/load testing as I'm very "pro test".  Some additional clarification (for both of us) might help make sure we're all working from the same playbook.

**Comment by Maralsotoudehnia** [comment]
@jefkel happy to chat, of course. Yes, I welcome additional clarification, especially surrounding which playbook to adhere to, responsibility, and purview.

**Comment by caggles** [comment]
Hi @Maralsotoudehnia @sbarre-esit @jefkel ! Based on Jeff's last comment, it seems like I don't need to take action on this - is that accurate? If so, please let me know if you'd like to leave the ticket open for further discussion. I'm totally fine with that, I just want to know if we should be keeping an eye on it to know if we'll need to action something :)

**Comment by sbarre-esit** [comment]
I'd like to keep the ticket open until the temporary increase is reverted.

**Comment by caggles** [comment]
No problem, I'll just move it out of our queue then :)

**Comment by Maralsotoudehnia** [comment]
@sbarre-esit can we please keep the temporary increase until November 15 2020? We're supporting critical business with needs that remain undefined and emerging and need to ensure we don't hit CPU limits again over the next month and a half. Happy to chat if there are any questions. Thank you in advance. cc @matthieu-foucault @Dianadec

**Comment by sbarre-esit** [comment]
No objections from me.

**Comment by Maralsotoudehnia** [comment]
Much appreciated, @sbarre-esit!!! Thank you :)

**Comment by caggles** [comment]
It's now past the Nov 15th deadline! @Maralsotoudehnia and @sbarre-esit - has the change been reverted so I can close this ticket?

**Comment by sbarre-esit** [comment]
I haven't made any changes yet, and https://isitreadonlyfriday.com/ == TRUE, so I'll change it on Monday unless there are any objections.

**Comment by matthieu-foucault** [comment]
Please don't change the quota on Monday. We currently use 83% of our CPU Limit quota in our production namespace.
We'll need a bit of time to look at our sysdig monitoring to see where it can be safely reduced. 
One one hand, the initial performance issue that triggered this request was reduced, but on the other hand, more HA (patroni) databases were deployed since the increase (that's where the CPU limits quota is used the most, as discussed in a meeting with Platform Services a while ago)

**Comment by Maralsotoudehnia** [comment]
@caggles @sbarre-esit @matthieu-foucault +1 to Matthieu's comment -- we're still having to support our business area with this extra quota and would really appreciate a bit more time. Please feel free to follow up with me if you have any questions. Best/m

**Comment by sbarre-esit** [comment]
Objection noted.

Sounds like this might be moving to a non-temporary change?

**Comment by jefkel** [comment]
I would suggest that the requester needs to add either:
a) a new issue to target new quota increase (aka - non-temporary)
b) a specific date to remove the temporary increase (new?)

**Comment by Maralsotoudehnia** [comment]
@jefkel @caggles @sbarre-esit It's temporary, with specific date to remove increase by January 15 2021. Thanks! cc @matthieu-foucault

**Comment by caggles** [comment]
@Maralsotoudehnia are we still on to turn the quota back down on the 15th?

**Comment by matthieu-foucault** [comment]
Given the new estimate for Sysdig's availability in the Silver cluster (which is required for us to migrate our production applications) is for Jan 20th (see https://github.com/BCDevOps/developer-experience/issues/797#issuecomment-755524019), I think that a more appropriate deadline would be Jan 30th

**Comment by jefkel** [comment]
Sounds like this is not worth keeping this open - Jan 30 feels like a "we can revoke temporary increase in quota when we complete the migration to Silver"?

Recommend that requests for quota increases in Silver (if required) are not treated as temporary request.  (no need for the platform team to continue to track an open ticket)

**Comment by matthieu-foucault** [comment]
Sounds good to me. 
The permanent quota increases in Silver were already requested and applied, so if this quota in the Pathfinder namespace can be left as-is until the namespaces are deleted (hopefully by the end of the month or early february), then that is good with me.

**Comment by Maralsotoudehnia** [comment]
@caggles if we could please keep the increase until Jan 30th/whenever we are unblocked to complete migration to Silver, then that would be awesome and greatly appreciated. I should note that I've discussed this request with @mitovskaol, who has been wonderfully helpful and understanding. We've already requested permanent quota increase in Silver @jefkel.

**Comment by caggles** [comment]
i'm going to close this ticket because it sounds like we're just going to delete the namespace instead of reducing the quota again.

**Comment by caggles** [comment]
For future, it seems like providing temporary increases isn't a great idea - we should consider quota increases permanent (along with requiring all the usual justification for a permanent increase) regardless of whether the team plans to reduce usage again later. We should approach it this way in the future instead of allowing temp requests with reduced justification.
